aiida/tools/data/array/kpoints/legacy.py

```diff
@@ -342,6 +342,10 @@ def _num_points_from_coordinates(path, point_coordinates, kpoint_distance=None):
     explicit_kpoints = [tuple(point_coordinates[path[0][0]])]
     labels = [(0, path[0][0])]
 
+    if not all([_.is_integer() for _ in num_points if isinstance(_, (float, numpy.float64))]):
+        raise ValueError('Could not determine number of points as a whole number. num_points={}'.format(num_points))
```

Comment:
Can you just duplicate the related unit test, but provide the faulty input to cover this section?

Reply:
Looking into the code and how it works, I am unsure whether this case will ever arise. But I am also not 100 % sure it won't. I simply cannot determine the pathway through the code to reproduce a `num_points` that contains values that would not constitute integers in a float format.

I can either remove the check, or leave it in for "safety"?

In the end, the codecov patch issue is not relevant. Although, it did correctly find that this particular line is not checked, I guess.

Reply:
I can also remove the block from codecov completely?

Reply:
> Looking into the code and how it works, I am unsure whether this case will ever arise. But I am also not 100 % sure it won't. I simply cannot determine the pathway through the code to reproduce a `num_points` that contains values that would not constitute integers in a float format.

There are just too many paths to determine that reliably. I mean clearly this has been an implicit assertion up until this point. So my suggestion is to make that assertion explicit:

`assert all([_.is_integer() for _ in num_points if isinstance(_, (float, numpy.float64))])`

The advantages of this handling is that the code segment would be covered by tests, it clearly indicates to future developers what assumptions are made here, and would make it very easy to diagnose the problem, should a user unexpectedly trigger this bug.

Reply:
Ah, you wanted to change the lines here to an `assert`. That makes more sense to me 👍 Can do. I misunderstood your original comment to mean adding a test in our test suite.